verifier_test.go

```diff
@@ -95,33 +95,36 @@ func TestNewVerifier(t *testing.T) {
 			},
 		},
 		{
-			name:    "rsa key mismatch",
+			name:    "rsa invalid public key",
 			alg:     AlgorithmPS256,
 			key:     ecdsaKey,
-			wantErr: true,
+			wantErr: "PS256: invalid public key",
 		},
 		{
 			name:    "rsa key under minimum entropy",
 			alg:     AlgorithmPS256,
 			key:     rsaKeyLowEntropy,
-			wantErr: true,
+			wantErr: "RSA key must be at least 2048 bits long",
 		},
 		{
 			name:    "unknown algorithm",
 			alg:     0,
-			wantErr: true,
+			wantErr: "algorithm not supported",
 		},
 		{
 			name:    "bogus ecdsa public key (point not on curve)",
 			alg:     AlgorithmES256,
 			key:     ecdsaKeyPointNotOnCurve,
-			wantErr: true,
+			wantErr: "public key point is not on curve",
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := NewVerifier(tt.alg, tt.key)
-			if (err != nil) != tt.wantErr {
+			if err != nil && (err.Error() != tt.wantErr) {
+				t.Errorf("NewVerifier() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			} else if err == nil && (tt.wantErr != "") {
```

Comment:
nit:
```suggestion
			if (err != nil && err.Error() != tt.wantErr) || (err == nil && tt.wantErr != "") {
```

Reply:
@shizhMSFT: I had a similar thought , but then felt that the existing construct is more readable than what is been suggested in your comment, hence I preferred this style!

Reply:
I agree with Yogesh, this check is already a bit too convoluted, we need to keep it readable.  Maybe one day we can move to stretchr/testify.